lib/path/tests/default.nix

```diff
@@ -10,9 +10,16 @@
   # Random seed
   seed ? null,
 }:
+
 pkgs.runCommand "lib-path-tests" {
   nativeBuildInputs = with pkgs; [
-    nix
+    # The aws-sdk-cpp tests are flaky.  Since pull requests to
+    # staging cause nix to be rebuilt, this means that staging PRs
+    # end up getting false CI failures due to whatever is flaky in
+    # the AWS SDK tests.  Since none of our CI needs to (or should
+    # be able to) contact AWS S3, let's just omit it all.  Bonus:
+    # the tests build way faster.
+    (nix.override { withAWS = false; })
```

Comment:
We can avoid the duplication here by not trying to make this file usable on its own, requiring people to go through `release.nix` instead. So the `system` argument can be removed, the default for `pkgs` can be removed, and `nix` should be a new argument inherited from `release.nix`.

Reply:
> We can avoid the duplication

Okay, I refactored to avoid duplication


> not trying to make this file usable on its own, requiring people to go through `release.nix` instead

I don't necessarily object to that, but a policy change like that should not be mixed up with this PR which simply turns off `withAWS` for CI.

Reply:
As the original author of the file and probably only user of the path library, I think it's totally okay to introduce `nix` as an argument there. It's an internal file without any stability guarantee.